waspc/packages/deploy/src/providers/fly/cmd/cmd.ts

```diff
@@ -0,0 +1,48 @@
+import { $, cd } from 'zx'
+import { ClientCommonOps, CommonOps, ServerCommonOps } from '../helpers/CommonOps.js'
+import { buildDirExists, getCommandHelp, waspSays } from '../helpers/helpers.js'
+import * as tomlHelpers from '../helpers/tomlFileHelpers.js'
+import { executeFlyCommand } from '../index.js'
+import { CmdOptions, SERVER_CONTEXT_OPTION } from './CmdOptions.js'
+
+// Runs a command by copying down the project toml files, executing it, and copying it back up (just in case).
+// If the toml file does not exist, some commands will not run with additional args (e.g. -a <appname>).
+export async function cmd(flyctlArgs: string[], options: CmdOptions) {
+  waspSays(`Running ${options.context} command: flyctl ${flyctlArgs.join(' ')}`)
+
+  if (!buildDirExists(options.waspDir)) {
+    waspSays('Building your Wasp app...')
+    cd(options.waspDir)
+    await $`${options.waspExe} build`
+  }
+
+  const tomlFiles = tomlHelpers.getTomlFileInfo(options)
+  let commonOps: CommonOps
+
+  if (options.context === SERVER_CONTEXT_OPTION) {
+    commonOps = new ServerCommonOps(options.waspDir, tomlFiles)
+  } else {
+    commonOps = new ClientCommonOps(options.waspDir, tomlFiles)
+  }
+
+  await runFlyctlCommand(commonOps, flyctlArgs)
+}
+
+async function runFlyctlCommand(commonOps: CommonOps, flyctlArgs: string[]) {
```

Comment:
Ok, we could have a chat since it might be interesting to discuss it, but feel free to ignore this for this PR, since it would probably be more work than benefits we get (if we get any). Evertyhing works well at the moment so probably best not to reshuffle it with an uncertain goal.